persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java

```diff
@@ -0,0 +1,235 @@
+/*
+ *  Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.persist.models;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class to process and store migrate differences.
+ *
+ *
+ */
+public class ModelDifferences {
```

Comment:
```suggestion
public class MigrationProcessor {
```

Reply:
how about changing the class name to either `MigrationProcessor` or `MigrationManager`

Reply:
In that case is it okay to keep it in the model package?

Reply:
Let's move it to utils package.

Reply:
This was agreed to keep in the `models` package as it only acts as a DataHolder for `migrate` and renamed as `MigrationDataHolder`.